sdk/core/Azure.Core/src/RequestContent.cs

```diff
@@ -52,6 +57,41 @@ public abstract class RequestContent : IDisposable
         /// <returns>An instance of <see cref="RequestContent"/> that wraps a <see cref="ReadOnlySequence{T}"/>.</returns>
         public static RequestContent Create(ReadOnlySequence<byte> bytes) => new ReadOnlySequenceContent(bytes);
 
+        /// <summary>
+        /// Creates a RequestContent representing the UTF-8 Encoding of the given <see cref="string"/>/
+        /// </summary>
+        /// <param name="content">The <see cref="string"/> to use.</param>
+        /// <returns>An instance of <see cref="RequestContent"/> that wraps a <see cref="string"/>.</returns>
+        /// <remarks>The returned content represents the UTF-8 Encoding of the given string.</remarks>
+        public static RequestContent Create(string content) => Create(UTF8NoBomEncoding.GetBytes(content));
```

Comment:
@KrzysztofCwalina I suspect you intended that this would do a conversion to UTF-8 since that's what we wanted to send on the wire, but just checking.  Do you think it would be worthwhile to have an optional Encoding parameter and if it's null assume UTF-8 (for cases where say you want to send a string in UTF-16 or some other encoding as part of an "upload" operation or something?)